### PR TITLE
Add Rust tour

### PR DESCRIPTION
**Pull request by gvdongen**
I didn't split it up in the different parts yet. I first wanted to do a check of this

### CODE REVIEW

File: templates/python/example.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ async def greet(ctx: Context, req: GreetingRequest) -> Greeting:
     await ctx.sleep(timedelta(seconds=1))
     await ctx.run("send reminder", lambda: send_reminder(greeting_id))
 
+    ctx.ge
```
Comment (slinkydeveloper) [inline templates/python/example.py:19]: left over?

File: tutorials/tour-of-restate-rust/src/cart_object.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            .reserve()
+            .call()
+            .await?;
+
+        if reservation_success {
+            let mut tickets = ctx
+                .get::<Json<Vec<String>>>(TICKETS)
+                .await?
+                .unwrap_or_default()
+                .into_inner();
```
Comment (slinkydeveloper) [inline tutorials/tour-of-restate-rust/src/cart_object.rs:36]: this looks so bad. I wonder if i can make this api a tad bit better :(

File: tutorials/tour-of-restate-rust/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+use restate_sdk::prelude::*;
+
+#[tokio::main]
+async fn main() {
+    // To enable logging
+    tracing_subscriber::fmt::init();
+
+    HttpServer::new(
+        Endpoint::builder()
+            .bind(cart_object::CartObject::serve(cart_object::CartObjectImpl))
```
Comment (slinkydeveloper) [inline tutorials/tour-of-restate-rust/src/main.rs:16]: you should be able to simply to

```suggestion
            .bind(cart_object::CartObjectImpl.serve())
```

But you need to import the `CartObject` trait

**Review by slinkydeveloper — COMMENTED** [review]
looks good for now, i assume clippy did its job here :)

File: tutorials/tour-of-restate-rust/src/cart_object.rs (hunk 2 above)
Comment (gvdongen) [reply tutorials/tour-of-restate-rust/src/cart_object.rs:36]: yeah it's quite verbose

**Pull request merged**